src/site/content/en/blog/mailru-cwv/index.md

```diff
@@ -0,0 +1,290 @@
+
+{% Aside %}
+SSR was not used for the entire application, as [it can be an expensive process](https://malloc.fi/performance-cost-of-server-side-rendered-react-node-js) that can delay [Time to First Byte (TTFB)](https://web.dev/ttfb/), which [indirectly affects LCP](https://web.dev/optimize-lcp/#optimize-your-server). We decided to use SSR for crucial content that the user will see on initial render, and offload rendering of hidden content on the client. At the moment, Mail.ru is experimenting with [Islands Architecture](https://jasonformat.com/islands-architecture/) to see if further performance gains can be had.
+{% endAside %}
+
+After implementing SSR, the cause of regression in CLS metrics was discovered that initially went unnoticed: the news block was not inserted at the moment of rendering the first content on the page. There was a delay between the initial painting of the page and the insertion of news. This behaviour resulted in an ad skeleton shift, which worsened CLS metrics.
+ 
+Although Chrome's DevTools showed Layout Shift events, we couldn't find the reason for it at first. Though SSR itself wasn't the problem, it helped in discovering the solution later on. Fixing the code responsible for the painting delay improved layout stability of the news component.
+
+<figure class="w-figure">
+  {% Img src="image/dB6B4Sr8kaaT0KZujRBFC303oFR2/X5ZgXiNPGEqz6oOlvY5v.png", alt="Finding the news painting problem with JavaScript disabled.", width="800", height="443" %}
+  <figcaption>Finding the news painting problem with JavaScript disabled.</figcaption>
+</figure> 
+
+<figure class="w-figure">
+  {% Img src="image/dB6B4Sr8kaaT0KZujRBFC303oFR2/aXnxfjZCk4oZN8s6ag5o.png", alt="Fixing the news painting problem with JavaScript disabled.", width="800", height="448" %}
+  <figcaption>Fixing the news painting problem with JavaScript disabled.</figcaption>
+</figure> 
+
+Another effect SSR can have on CLS is the movement components before and after hydration. We encountered this on the mobile version in particular, which led to further layout shifts and negative impacts on CLS, which required special attention to be paid to hydrated component markup. One recurring solution to this problem was to transfer as much display logic from JavaScript to CSS whenever possible.
+
+## Code splitting and unused polyfills
+
+To improve the perceived page load speed, work was required to decrease LCP and FID values. One way to achieve this is through [code splitting](https://web.dev/codelab-code-splitting/). Our team, in addition to the Mail.ru home page itself, is developing a widget for portal navigation. It is currently embedded in [a large number of VK projects](https://vk.company/en/projects/).
```

Comment:
```suggestion
To improve the perceived page load speed, work was required to decrease LCP and FID values. One way to achieve this is through [code splitting](https://web.dev/codelab-code-splitting/). In addition to the Mail.ru home page itself, our team is developing a widget for portal navigation. It is currently embedded in [many of our parent company's projects](https://vk.company/en/projects/).
```

Reply:
This flows better, and is based on my reading of [this Wikipedia article](https://en.wikipedia.org/wiki/VK_(company)).